app/javascript/packages/request/index.ts

```diff
@@ -15,7 +15,10 @@ interface RequestOptions extends RequestInit {
 const getCSRFToken = () =>
   document.querySelector<HTMLMetaElement>('meta[name="csrf-token"]')?.content;
 
-export const request = async (url: string, options: Partial<RequestOptions> = {}) => {
+export async function request<Response>(
```

Comment:
We could consider making the generic optional by `<Response = any>`, but I think it might be a good constraint to require developers to provide the response shape. What do you think?